cluster/images/provider-terraform/Dockerfile

```diff
@@ -1,15 +1,26 @@
-FROM hashicorp/terraform:1.2.3
+FROM alpine:3.16.2
+RUN apk --no-cache add ca-certificates bash
 
 ARG TARGETOS
 ARG TARGETARCH
 
-ADD bin/$TARGETOS\_$TARGETARCH/provider /usr/local/bin/crossplane-terraform-provider
+ENV TERRAFORM_VERSION=1.3.4
+# Prevent terraform from forking in case of crash or panic, useful when debugging
+ENV TF_FORK 0
```

Comment:
```suggestion
```
Just double-checked that there is no `TF_FORK` in the v1.3.4 state  of https://github.com/hashicorp/terraform/

It is visible that it was removed in https://github.com/hashicorp/terraform/pull/29825 

Apparently it is outdated practice so we don't need it in Dockerfile either